scripts/oneoff/remove-dos4-questions-from-dos5-drafts.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+
+import argparse
+import sys
+
+sys.path.insert(0, '.')
+from dmapiclient import DataAPIClient
+from dmscripts.helpers.auth_helpers import get_auth_token
+from dmutils.env_helpers import get_api_endpoint_from_stage
+
+from typing import List
+
+
+def get_affected_drafts_services(api_client: DataAPIClient) -> List[dict]:
+    all_drafts = api_client.find_draft_services_by_framework('digital-outcomes-and-specialists-5')["services"]
+
+    return [draft for draft in all_drafts if draft_service_contains_dos4_answer(draft)]
+
+
+def draft_service_contains_dos4_answer(draft: dict) -> bool:
+    invalid_answers = {
+        "agileCoachPriceMin",
+        "businessAnalystPriceMin",
+        "communicationsManagerPriceMin",
+        "contentDesignerPriceMin",
+        "dataArchitectPriceMin",
+        "dataEngineerPriceMin",
+        "dataScientistPriceMin",
+        "deliveryManagerPriceMin",
+        "designerPriceMin",
+        "developerPriceMin",
+        "performanceAnalystPriceMin",
+        "portfolioManagerPriceMin",
+        "productManagerPriceMin",
+        "programmeManagerPriceMin",
+        "qualityAssurancePriceMin",
+        "securityConsultantPriceMin",
+        "serviceManagerPriceMin",
+        "technicalArchitectPriceMin",
+        "userResearcherPriceMin",
+        "webOperationsPriceMin",
+        "developerAccessibleApplications",
+        "contentDesignerAccessibleApplications",
+        "qualityAssuranceAccessibleApplications",
+        "technicalArchitectAccessibleApplications",
+        "webOperationsAccessibleApplications",
+        "serviceManagerAccessibleApplications",
+        "designerAccessibleApplications"
+    }
+
+    return len(invalid_answers.union(set(draft.keys()))) > 0 and draft['status'] == 'not-submitted'
```

Comment:
I like the use of sets here! I'm not sure you need to do `set(draft.keys())` though, because drafts should be a `set`  already 😁  [[1](https://renzolucioni.com/pythons-set-literals/)]

Oh and while we're at it, the use of `len` can also be removed, because any non-empty set is truthy [[2](https://docs.python.org/3.6/library/stdtypes.html#truth-value-testing)].

---

However, I'm not sure the union is what we want here?

> Return a new set with elements from the set and all others.
https://docs.python.org/3.6/library/stdtypes.html#frozenset.union

Reply:
Ah thanks, I didn't realise that passing a dict into `set` would just take the keys and good point on the truthiness too

You're right, union is definitely wrong 😂  We want the set of values that are common to both sides which is intersection